### PR TITLE
feat: support initial values

### PR DESCRIPTION
**Pull request by Silhoue**
## Summary

Allow defining `initialValue` property on content type fields. Validate for supported field types, existing locales and correct type of initial values.

### CODE REVIEW

File: src/lib/offline-api/validator/content-type.ts [hunk 1]
```diff
@@ -2,7 +2,14 @@ import { ApiHook } from '../'
...
+  contentType: ContentType
+  locales: string[]
+  savedContentType?: ContentType
+  publishedContentType?: ContentType
+}
+
 export interface ContentTypePayloadValidator {
   hooks: ApiHook[]
-  validate (contentType: ContentType, savedContentType?: ContentType, publishedContentType?: ContentType): (InvalidActionError | PayloadValidationError)[]
+  validate (options: ContentTypePayloadValidatorOptions): (InvalidActionError | PayloadValidationError)[]
```
Comment (zcei) [inline src/lib/offline-api/validator/content-type.ts:14]: Hugs for moving this away from positional args!

File: src/lib/offline-api/validator/errors.ts [hunk 2]
```diff
@@ -64,6 +64,21 @@ const errors = {
       INVALID_VALIDATION_PARAMETER: (propName, expectedType, actualType) => {
         return `"${propName}" validation expected to be "${expectedType}", but got "${actualType}"`
       }
+    },
+    initialValue: {
+      TYPE_MISMATCH: (fieldId, valueType, locale, fieldType) => {
```
Comment (zcei) [inline src/lib/offline-api/validator/errors.ts:69]: Should we consider validations here as well? Like when you allow a list of values or numbers within a range and create some sort of `VALIDATED_TYPE_MISMATCH` or is this too much effort to justify its usefulness?

**Review by zcei — COMMENTED** [review]
Looks really good! 💪

File: test/unit/lib/offline-api/validation/payload-validation-field-initialvalue.spec.ts [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+        const lunch = migration.createContentType('lunch').name('lunch')
+        lunch.createField('mainCourse')
+        .name('mainCourse')
+        .type('Symbol')
+        .initialValue({
+          'en-US': 'A Symbol',
+          'de-DE': 'A Symbol',
+          'fr-FR': 'A Symbol'
+        })
+      }, [], [], [], [ 'en-US'])
```
Comment (TimBeyer) [inline test/unit/lib/offline-api/validation/payload-validation-field-initialvalue.spec.ts:20]: All of the locale arrays have a space too much in the beginning. We really need to fix the linter setup here 😄

Comment (Silhoue) [reply test/unit/lib/offline-api/validation/payload-validation-field-initialvalue.spec.ts:20]: Added [DANTE-102](https://contentful.atlassian.net/browse/DANTE-102) to our backlog.

File: src/lib/offline-api/validator/errors.ts (hunk 2 above)
Comment (Silhoue) [reply src/lib/offline-api/validator/errors.ts:69]: We currently allow saving a content type with invalid initial field values 🙂

Comment (zcei) [reply src/lib/offline-api/validator/errors.ts:69]: Oh okay, so we do rely on the entry validations later on 💡 . 

> Nothing to see here, keep walking folks..

**Review by zcei — APPROVED** [review]

**Pull request merged**